### DISCUSSION TITLE
table 功能请求 能不能直接在表里面弄个字段,把table 的设置用JSON全写里面 和示例里面的json一样

### DISCUSSION THREAD

**Discussion by aimu2000** [layui/layui#1655] [讨论]
### 议题条件

- [X] 我确认已查看官方使用文档：https://layui.dev ，但没有找到相关解决方案。
- [X] 我确认已在 [Issues](https://github.com/layui/layui/issues?q=is%3Aissue ) 中搜索过类似的问题，但没有找到相关解决方案。
- [X] 我已仔细阅读： 🍀[ Layui Issue 贡献指南](https://github.com/layui/layui/blob/main/CONTRIBUTING.md)

### 议题类型

功能请求

### 使用版本

2.9.6

### 问题描述

table 功能请求 直接在表里面加个字段,把table 的设置用JSON全写里面 和示例里面的json一样 把 cols:也写在里面.所有的都打包存SQL.直接读SQL就可,table.render({
    elem: '#test',
    sql:sql
        // 省略这里
    done: function(){

### 业务代码

```auto
table.render({
    elem: '#test',
    url: '/static/json/2/table/demo1.json', // 此处为静态模拟数据，实际使用时需换成真实接口

    toolbar: '#toolbarDemo',
    defaultToolbar: ['filter', 'exports', 'print', {
      title: '提示',
      layEvent: 'LAYTABLE_TIPS',
      icon: 'layui-icon-tips'
    }],
```

### 浏览器

ALL

### 友好承诺

- [X] 我承诺将本着相互尊重、理解和友善的态度进行交流，共同维护 Layui 良好的社区氛围。

**Comment by sentsim** [comment]
组件接受的 options 包含了各种回调，当然不能全部以 JSON 的形式接受。
若要实现你提到的动态配置，你也可以在 `table.render()` 执行之前，先请求配置接口，将获得的配置信息赋值给组件。

**Comment by LSL1618** [comment]
理论上是可以的，`table.render(options)`中的options本质上就是一个JSON对象，把它转成字符串存到表字段或者配置文件里都可以，只要把它获取到再转成JSON对象传递给render方法就可以实现动态配置。
